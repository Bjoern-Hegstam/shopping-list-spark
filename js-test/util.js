import { shallow } from 'enzyme/build';
import React from 'react';

export function setupComponent(Component, defaultProps, optProps) {
    const props = {
        ...defaultProps,
        ...optProps,
    };

    const component = shallow(<Component {...props} />);

    return { component, props };
}
