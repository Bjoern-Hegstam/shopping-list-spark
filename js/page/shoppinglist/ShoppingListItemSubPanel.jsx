import * as PropTypes from 'prop-types';
import React from 'react';

export default function ShoppingListItemSubPanel({ onIncrementClick, onDecrementClick }) {
  return (
    <div className="shopping-list-item-sub-panel">
      <div
        className="shopping-list-item-sub-panel__button shopping-list-item-sub-panel__button__incr"
        onClick={onIncrementClick}
      >
        <span>+</span>
      </div>
      <div
        className="shopping-list-item-sub-panel__button shopping-list-item-sub-panel__button__decr"
        onClick={onDecrementClick}
      >
        <span>-</span>
      </div>
    </div>
  );
}

ShoppingListItemSubPanel.propTypes = {
  onIncrementClick: PropTypes.func.isRequired,
  onDecrementClick: PropTypes.func.isRequired,
};
