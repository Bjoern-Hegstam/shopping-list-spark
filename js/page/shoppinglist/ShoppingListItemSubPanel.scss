@import '../../styles/general';

.shopping-list-item-sub-panel {
  & {
    grid-area: sub-panel;

    width: 100%;
    height: 100%;

    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-areas: "incr decr";

    background-color: darken($background-color, 10%);
  }

  &__button {
    @extend %base-button;
    font-weight: normal;
    font-size: 2.5rem;
    color: black;

    padding: 0;
    border-radius: 0;

    &:hover {
      background-color: darken($background-color, 20%);
    }

    &__incr {
      grid-area: incr;

      border-right: 1px solid black;
    }

    &__decr {
      grid-area: decr;
    }
  }
}
