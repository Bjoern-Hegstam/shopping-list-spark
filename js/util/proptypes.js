import PropTypes from 'prop-types';

export const ShoppingListType = PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    items: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        itemType: PropTypes.shape({
            id: PropTypes.string.isRequired,
            name: PropTypes.string.isRequired
        }).isRequired,
        quantity: PropTypes.number.isRequired,
        inCart: PropTypes.bool.isRequired
    })).isRequired
});