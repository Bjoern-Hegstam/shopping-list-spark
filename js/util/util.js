export const hasId = id => item => item.id === id;
export const hasRequestId = requestId => item => item.requestId === requestId;
export const not = predicate => params => !predicate(params);
