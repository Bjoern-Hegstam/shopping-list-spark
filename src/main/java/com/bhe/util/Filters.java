package com.bhe.util;

import com.bhe.login.LoginController;
import spark.Request;
import spark.Response;

import static spark.Spark.halt;

public class Filters {
    public static void userIsLoggedIn(Request request, Response response) {
        boolean userIsLoggedIn = LoginController.userIsLoggedIn(request);
        if (!userIsLoggedIn) {
            halt(401, "Not authorized");
        }
    }
}
