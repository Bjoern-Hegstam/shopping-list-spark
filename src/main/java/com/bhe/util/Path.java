package com.bhe.util;

public class Path {

    public static class Web {
        public static final String INDEX = "/";
        public static final String LOGIN = "/login";
        public static final String LOGOUT = "/logout";
    }

    public static class Template {
        public static final String LOGIN = "/velocity/login/login.vm";
        public static final String INDEX = "/velocity/index/index.vm";
    }
}
