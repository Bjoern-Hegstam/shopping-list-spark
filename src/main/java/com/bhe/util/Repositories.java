package com.bhe.util;

import com.bhe.user.UserRepository;
import com.bhe.user.UserRepositoryInMem;

public class Repositories {
    private static UserRepository userRepository;

    public static void initDev() {
        userRepository = new UserRepositoryInMem();
    }

    public static UserRepository users() {
        return userRepository;
    }
}
