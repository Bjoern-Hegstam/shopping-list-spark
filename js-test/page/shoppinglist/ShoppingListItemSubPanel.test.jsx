import { setupComponent } from '../../util';
import ShoppingListItemSubPanel from '../../../js/page/shoppinglist/ShoppingListItemSubPanel';

function setup(optProps) {
  return setupComponent(
    ShoppingListItemSubPanel,
    {
      onIncrementClick: jest.fn(),
      onDecrementClick: jest.fn(),
    },
    optProps,
  );
}

let component;
let props;

beforeEach(() => {
  ({ component, props } = setup());
});

it('initial render', () => {
  expect(component.debug()).toMatchSnapshot();
});

it('calls onIncrementClick', () => {
  component.find('.shopping-list-item-sub-panel__button__incr').simulate('click');

  expect(props.onIncrementClick).toHaveBeenCalledTimes(1);
  expect(props.onDecrementClick).toHaveBeenCalledTimes(0);
});

it('calls onDecrementClick', () => {
  component.find('.shopping-list-item-sub-panel__button__decr').simulate('click');

  expect(props.onIncrementClick).toHaveBeenCalledTimes(0);
  expect(props.onDecrementClick).toHaveBeenCalledTimes(1);
});
