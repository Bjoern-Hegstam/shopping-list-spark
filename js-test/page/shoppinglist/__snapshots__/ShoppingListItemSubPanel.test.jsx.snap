// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial render 1`] = `
"<div className=\\"shopping-list-item-sub-panel\\">
  <div className=\\"shopping-list-item-sub-panel__button shopping-list-item-sub-panel__button__incr\\" onClick={[Function: mockConstructor]}>
    <span>
      +
    </span>
  </div>
  <div className=\\"shopping-list-item-sub-panel__button shopping-list-item-sub-panel__button__decr\\" onClick={[Function: mockConstructor]}>
    <span>
      -
    </span>
  </div>
</div>"
`;
